File: database/jipson7/sortingComparison-ScalaVsJava/sortingComparison-master/clojure/bubblesort/sort.clj
(ns sorting)

(defn bubble-swap [the-list]
  (if (or (nil? the-list) (nil? (second the-list)))
    the-list
    (if (cust-comp (first the-list) (second the-list))
      (cons (second the-list) (cons (first the-list) (nthrest the-list 2)))
      (cons (first the-list) (bubble-swap (rest the-list))))))

(defn bubble-sort [a-list]
  (if (= a-list (bubble-swap a-list))
    a-list
    (recur (bubble-swap a-list))))
 

File: database/jipson7/sortingComparison-ScalaVsJava/sortingComparison-master/scala/bubblesort/Sort.scala
package bubblesort {


  object Sort {

    def BubbleSort[T](list:Array[T], sort:(T, T)=>Boolean):Array[T] = {

      val length = list.size - 1 

      for (i <- 1 to length) {

        for (j <- length to i by -1) {

          if (sort(list(j), list(j - 1))) {

            var holder = list(j)
            list(j) = list(j - 1)
            list(j - 1) = holder


          }

        }

      }

      list

    }

  }

}
